src/server/scripts/EasternKingdoms/ZulGurub/boss_marli.cpp

```diff
@@ -77,142 +80,180 @@ enum Misc
 float const DamageIncrease = 35.0f;
 float const DamageDecrease = 100.f / (1.f + DamageIncrease / 100.f) - 100.f;
 
+float const MaxRangeHatch = 100.f;
+
 class boss_marli : public CreatureScript
 {
     public: boss_marli() : CreatureScript("boss_marli") { }
 
         struct boss_marliAI : public BossAI
         {
-            boss_marliAI(Creature* creature) : BossAI(creature, DATA_MARLI) { }
-
-            void Reset() override
-            {
-                if (events.IsInPhase(PHASE_THREE))
-                    me->ApplyStatPctModifier(UNIT_MOD_DAMAGE_MAINHAND, TOTAL_PCT, DamageDecrease); // hack
-                _Reset();
-            }
+            public:
+                boss_marliAI(Creature* creature) : BossAI(creature, DATA_MARLI) { }
 
-            void JustDied(Unit* /*killer*/) override
-            {
-                _JustDied();
-                Talk(SAY_DEATH);
-            }
+                void Initialize()
+                {
+                    _hatchedEggs = std::set<GameObject*>();
```

Comment:
is this line needed ?

Reply:
I think in theory no, but I've read [somewhere](https://drive.google.com/file/d/0Bw08_9WDSwhKVHBRT0dNc0dZQ2c/view) that it is good practice and convention in TC to initialize class members like that.

But I can delete it I am also not a fan.

Reply:
Not needed for types that have a default constructor (set obviously does)